app/src/main/java/com/droidhats/campuscompass/views/FloorFragment.kt

```diff
@@ -68,23 +293,255 @@ class FloorFragment : Fragment() {
         val canvas = Canvas(bitmap)
         canvas.drawARGB(0, 255, 255, 255)
         svg.renderToCanvas(canvas)
+        if (cachedSVG != null) prevSVG = cachedSVG
+        cachedSVG = svg
+        imageView.visibility = View.VISIBLE
         imageView.setImageDrawable(BitmapDrawable(getResources(), bitmap))
     }
 
-    override fun onActivityCreated(savedInstanceState: Bundle?) {
-        super.onActivityCreated(savedInstanceState)
-        viewModel = ViewModelProvider(this).get(FloorViewModel::class.java)
+    fun handleNavigation(startToEnd: Pair<IndoorLocation, IndoorLocation>) {
 
-        initSearchBar()
+        val indoorInstructionsLayout: LinearLayout = root.findViewById(R.id.indoorInstructionsLayout)
+        indoorInstructionsLayout.visibility = View.VISIBLE
 
-        val startAndEnd = viewModel.getDirections()
-        if (startAndEnd != null) {
-            val inputStream: InputStream = requireContext().assets.open("hall8.svg")
-            val file: String = inputStream.bufferedReader().use { it.readText() }
-            val mapProcessor: ProcessMap = ProcessMap()
-            mapProcessor.readSVGFromString(file)
-            val svg: SVG = SVG.getFromString(mapProcessor.getSVGStringFromDirections(startAndEnd))
-            setImage(svg)
+        var startAndEnd = startToEnd
+        var building: Building
+        if (startAndEnd.first.lID == "") {
+            building = viewModelMapViewModel.getBuildings()[startAndEnd.second.buildingIndex]
+        } else {
+            building = viewModelMapViewModel.getBuildings()[startAndEnd.first.buildingIndex]
+        }
+
+        if (
+            (startAndEnd.first.lID == "" && startAndEnd.second.floorNum != "1")
+            || (startAndEnd.second.lID == "" && startAndEnd.first.floorNum != "1")
+            || (startAndEnd.first.floorNum != startAndEnd.second.floorNum
+                    && startAndEnd.second.lID != "" && startAndEnd.first.lID != "")
+        ) {
+            canConsume = false
+            hasTwoSteps = true
+        }
+
+        val goingUp: Boolean = when(true) {
+            (startAndEnd.first.lID != "" && startAndEnd.second.lID != "") -> {
+                startAndEnd.first.getFloorNumber() < startAndEnd.second.getFloorNumber()
+            }
+            startAndEnd.first.lID == "" -> {
+                1 < startAndEnd.second.getFloorNumber()
+            }
+            else -> { // second.lID is an empty string
+                startAndEnd.first.getFloorNumber() < 1
+            }
+        }
+
+
```

Comment:
add comments here to explain the onclick listener methods

Reply:
Done